### PR TITLE
sequoia-chameleon-gnupg: init at 0.1.1

### PR DESCRIPTION
**Pull request by NickCao**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/sequoia-chameleon-gnupg/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    hash = "sha256-msJOQUqq0S+cttSucvXxAFIcFu3R4lUCgIpkaumkaZY=";
+  };
+
+  cargoHash = "sha256-rnZPZY4mg18Plwcrppn7a1+H6CI/A4ntQ5TsZvi6kcA=";
+
+  nativeBuildInputs = [ rustPlatform.bindgenHook pkg-config ];
+  buildInputs = [ nettle ];
+
+  # gpgconf: error creating socket directory
+  doCheck = false;
```
Comment (NickCao) [inline pkgs/tools/security/sequoia-chameleon-gnupg/default.nix:25]: I just can't make the tests work, to reproduce the issue:
```nix
runCommand "test" {
  nativeBuildInputs = [ gnupg ];
} "HOME=$TMPDIR gpgconf --create-socketdir"
```

File: pkgs/tools/security/sequoia-chameleon-gnupg/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  meta = with lib; {
+    description = "Sequoia's reimplementation of the GnuPG interface";
+    homepage = "https://sequoia-pgp.org/";
```
Comment (doronbehar) [inline pkgs/tools/security/sequoia-chameleon-gnupg/default.nix:29]: Maybe be a bit more specific?

```suggestion
    homepage = "https://gitlab.com/sequoia-pgp/sequoia-chameleon-gnupg";
```

File: pkgs/tools/security/sequoia-chameleon-gnupg/default.nix (hunk 1 above)
Comment (doronbehar) [reply pkgs/tools/security/sequoia-chameleon-gnupg/default.nix:25]: I think sockets don't work in Nix' sandbox, it's not so bad to disable tests due to this. We can still maintain this package.

Comment (doronbehar) [reply pkgs/tools/security/sequoia-chameleon-gnupg/default.nix:25]: @NickCao would you like to keep things as are regarding the tests?

Comment (NickCao) [reply pkgs/tools/security/sequoia-chameleon-gnupg/default.nix:25]: I don't have the time to find a fix these days, I'd leave it for later.

Comment (doronbehar) [reply pkgs/tools/security/sequoia-chameleon-gnupg/default.nix:25]: I think we can merge this as is in the meantime.

**Pull request merged**